[FAB-15648] hints to document: Non-TLS orderer with etcdraft usage

davidkhala opened the pull request:
#### Type of change

- Documentation update

#### Description

- Notes about configure TLSCARoot
- Notes about configure non-general-TLS orderer with etcdraft mode




#### Additional details

Signed-off-by: davidliu <david-khala@hotmail.com>

#### Related issues

FAB-15648

mastersingh24 commented on docs/source/raft_configuration.md:
```diff
@@ -86,22 +86,28 @@ other, and TLS certificates issued by a public TLS CA for the client facing API.
 be either set together or unset together.
 If they are unset, they are inherited from the general TLS section,
 in example `general.tls.{privateKey, certificate}`.
+In the use case when general TLS is disabled, these parameters must be set while:
```
```suggestion
In the use case when general TLS is disabled, these parameters must be set:
```
small typo I think

davidkhala committed:
[FAB-15648] document update: Non-TLS orderer with etcdraft usage
- Notes about configure TLSCARoot
- Notes about configure non-general-TLS orderer with etcdraft mode

Signed-off-by: davidliu <david-khala@hotmail.com>

wlahti commented on docs/source/raft_configuration.md:
```diff
@@ -86,22 +86,28 @@ other, and TLS certificates issued by a public TLS CA for the client facing API.
-  * `ListenPort`: The port the cluster listens on. If blank, the port is the same
-  port as the orderer general port (`general.listenPort`)
+  * `ListenPort`: The port the cluster listens on. 
+  It must be same as `consenters[i].Port` in Channel configuration. 
```
Not sure if this is the best place to mention this. It feels like this belongs in the section describing the Consenters but that would require reordering some sections of this doc. The natural flow to me would be to describe the configuration of a single raft node and then how it fits into the channel configuration, not vice versa.

davidkhala replied:
Agree, I see this difficulty in how to express how they are bound. If another maintainer also agree on we are better to reorganize sections then I am glad to change accordingly.

wlahti commented on docs/source/raft_configuration.md:
```diff
@@ -86,22 +86,28 @@ other, and TLS certificates issued by a public TLS CA for the client facing API.
 be either set together or unset together.
 If they are unset, they are inherited from the general TLS section,
 in example `general.tls.{privateKey, certificate}`.
+In the use case when general TLS is disabled, these parameters must be set:
```
```suggestion
When general TLS is disabled:
```

wlahti commented on docs/source/raft_configuration.md:
```diff
@@ -86,22 +86,28 @@ other, and TLS certificates issued by a public TLS CA for the client facing API.
 If they are unset, they are inherited from the general TLS section,
 in example `general.tls.{privateKey, certificate}`.
+In the use case when general TLS is disabled, these parameters must be set:
+ - use different `ListenPort` than the orderer general port
```
```suggestion
 - use a different `ListenPort` than the orderer general port
```

wlahti commented on docs/source/raft_configuration.md:
```diff
@@ -86,22 +86,28 @@ other, and TLS certificates issued by a public TLS CA for the client facing API.
 in example `general.tls.{privateKey, certificate}`.
+In the use case when general TLS is disabled, these parameters must be set:
+ - use different `ListenPort` than the orderer general port
+ - Properly configure TLS root CAs. Search scope includes
```
Why would we need to configure the TLS root CAs when TLS is disabled? I've never tried to use the orderer without TLS enabled but this seems odd.

jyellick replied:
I imagine this is if you aren't defining the TLS CAs in the channel config? Although seems better to define them in the channel config even if TLS is disabled generally.

davidkhala replied:
I see consenters mutual connection errors when TLS root CAs could not be found in anywhere. I am not 100% sure of the search scope, but indeed we should have at least once.

davidkhala replied:
How about re-statement as  `Properly configure TLS root CAs. For instance, define them for each orderer organization in channel config ` @jyellick

jyellick replied:
Sorry for losing track of this.

Yes, that sentence is better.  Though I think it would be superior to say "Properly configure TLS root CAs in the channel configuration."  I do not believe that recent versions of Fabric will even allow you to define Raft endpoints if their TLS certificate is not valid according to some TLS CA in the channel config.

caod123 commented:
It looks like this PR hasn't been touched in some time. @jyellick can you review @davidkhala's last question?

davidkhala commented:
@jyellick update and look for review again

pamandrejko commented:
@Mergifyio backport release-2.3

pamandrejko commented:
@Mergifyio backport release-2.2
